include/openPMD/RecordComponent.hpp

```diff
@@ -114,6 +116,14 @@ class RecordComponent : public BaseRecordComponent
      *
      * If offset is non-zero and extent is {-1u} the leftover extent in the
      * record component will be selected.
+     *
+     * In FlushType::DEFER, the data managed by the returned
+     * std::shared_ptr< T > MUST NOT be read, modified or deleted until
```

Comment:
Modification would probably not even be detrimental, but rather useless.

Reply:
Yes, I basically just describe the contract here. Both is not allowed from a user-side for various reasons (either ignored or even crashes).